### ISSUE TITLE
Insert method suggestion

### ISSUE THREAD

**Issue by enumag** [driftphp/reactphp-dbal#10]
While implementing something similar I added my own insert implementation because I wasn't satisfied with [yours](https://github.com/driftphp/reactphp-dbal/blob/fe22c18a9523beeb38d5194a0b8ff64fa4cfcb4a/src/Connection.php#L162-L174). The point is that it feels unnatural to pass `$values` and `$parameters` separately. I'm aware it is more flexible since you can use `NOW()`, subqueries and what not but in reality I wouldn't mind using the QueryBuilder directly for such cases. I see the insert method as just a simple helper so I wanted API like this:

```php
$connection->insert($table, ['column' => 'value', ...]);
```

It can be implemented quite easily:

```php
	public function insert(string $table, array $values): PromiseInterface
	{
		$queryBuilder = $this->createQueryBuilder();
		$queryBuilder->insert($table);

		foreach ($values as $column => $value) {
			$queryBuilder
				->setValue($column, ':' . $column)
				->setParameter($column, $value);
		}

        return $this->query($queryBuilder);
	}
```

It might be nice to add an `update` and `delete` methods with similar API:

```php
	public function update(string $table, array $values, array $where): PromiseInterface;
	public function delete(string $table, array $where): PromiseInterface;
```

Note: I don't really care if you use this or not since I'm not using Drift directly. It's just a suggestion. Use it or close it as you see fit. ;-)

Btw. [this](https://github.com/driftphp/reactphp-dbal/blob/master/src/Connection.php#L177) is incorrect since it's a delete method.

**Comment by mmoreram** [comment]
I'm working on it right now.

Thanks for the suggestions.

**Comment by enumag** [comment]
Closed by #11
